pynxtools/dataconverter/helpers.py

```diff
@@ -456,15 +459,28 @@ def ensure_all_required_fields_exist(template, data, nxdl_root):
             opt_parent = check_for_optional_parent(path, nxdl_root)
             if opt_parent != "<<NOT_FOUND>>":
                 if does_group_exist(opt_parent, data) and not does_group_exist(renamed_path, data):
-                    raise ValueError(f"The required group, {path}, hasn't been supplied"
-                                     f" while its optional parent, {path}, is supplied.")
+                    missing_groups_and_fields['groups'].appends(path)
```

Comment:
I would just use logging here and do a `logger.warning`.

Reply:
So, you do not want the converter blocked, even the required fields are not rendered.

Reply:
This is what I had in mind. No need to collect all these errors like this. We can just output the warnings.